src/retro_achievements.cpp

```diff
@@ -0,0 +1,1653 @@
+            {
+                sg_destroy_image(atlas->image);
+            }
+            atlas->image.id = SG_INVALID_ID;
+        }
+
+        if (atlas->image.id == SG_INVALID_ID)
+        {
+            sg_image_desc desc = {0};
+            desc.type = SG_IMAGETYPE_2D, desc.render_target = false,
+            desc.width = atlas->pixel_stride, desc.height = atlas->pixel_stride,
+            desc.num_slices = 1, desc.num_mipmaps = 1, desc.usage = SG_USAGE_DYNAMIC,
+            desc.pixel_format = SG_PIXELFORMAT_RGBA8, desc.sample_count = 1,
+            desc.min_filter = SG_FILTER_LINEAR, desc.mag_filter = SG_FILTER_LINEAR,
+            desc.wrap_u = SG_WRAP_CLAMP_TO_EDGE, desc.wrap_v = SG_WRAP_CLAMP_TO_EDGE,
+            desc.wrap_w = SG_WRAP_CLAMP_TO_EDGE,
+            desc.border_color = SG_BORDERCOLOR_TRANSPARENT_BLACK, desc.max_anisotropy = 1,
+            desc.min_lod = 0.0f, desc.max_lod = 1e9f,
+
+            atlas->image = sg_make_image(&desc);
+
+            if (atlas->resized)
+            {
+                for (auto& image : ra_state->game_state->image_cache)
+                {
+                    // Update the images to point to the new atlas instead
+                    if (image.second.width == atlas->tile_width &&
+                        image.second.height == atlas->tile_height)
+                    {
+                        image.second.atlas_id = atlas->image.id;
+                    }
+                }
+            }
+        }
+
+        if (atlas->dirty)
+        {
+            sg_image_data data = {0};
+            data.subimage[0][0].ptr = atlas->data.data();
+            data.subimage[0][0].size = atlas->data.size();
+            printf("updating image %d\n", atlas->image.id);
+            sg_update_image(atlas->image, data);
+        }
+
+        atlas->dirty = false;
+        atlas->resized = false;
+    }
+}
+
+void retro_achievements_draw_panel(int win_w)
+{
+    const rc_client_user_t* user = rc_client_get_user_info(ra_state->rc_client);
+    igPushIDStr("RetroAchievements");
+    if (!user)
+    {
+        static char username[256] = {0};
+        static char password[256] = {0};
+        bool pending_login = ra_state->pending_login;
+        se_text("Username");
+        igSameLine(win_w - 150, 0);
+        if (pending_login)
+            se_push_disabled();
+        bool enter = igInputText("##Username", username, sizeof(username),
+                                 ImGuiInputTextFlags_EnterReturnsTrue, NULL, NULL);
+        if (pending_login)
+            se_pop_disabled();
+        se_text("Password");
+        igSameLine(win_w - 150, 0);
+        if (pending_login)
+            se_push_disabled();
+        enter |= igInputText("##Password", password, sizeof(password),
+                             ImGuiInputTextFlags_Password | ImGuiInputTextFlags_EnterReturnsTrue,
+                             NULL, NULL);
+        if (se_button(ICON_FK_SIGN_IN " Login", ImVec2{0, 0}) || enter)
```

Comment:
Can you add a register button that opens up a browser to: https://retroachievements.org/createaccount.php

I think we could reuse the code that opens up the popups for the cloud drive logins.